lib/js/src/manager/screen/choiceset/_PreloadPresentChoicesOperation.js

```diff
@@ -0,0 +1,824 @@
+     * @param {Boolean} isVrOptional - Whether core doesn't require VR commands
+     * @param {ChoiceCell[]} cellsToPreload - Upload these
+     * @param {ChoiceCell[]} loadedCells - Cells that have already been uploaded
+     * @param {ChoiceSet} choiceSet - The choice set to present.
+     * @param {InteractionMode} presentationMode - How the interaction is presented in the HMI
+     * @param {KeyboardProperties} originalKeyboardProperties - The keyboard properties
+     * @param {KeyboardListener} keyboardListener - For listening to keyboard events
+     * @param {ChoiceSetSelectionListener} choiceSetSelectionListener - A callback function for selection updates
+     * @param {Number} cancelId - The ID used to cancel the operation
+     * @param {function} completionListener - A callback function for operation updates. Passes back true or false for successful operation.
+     */
+    constructor (lifecycleManager = null, fileManager = null, displayName = null,
+        defaultMainWindowCapability = null, isVrOptional, cellsToPreload = [],
+        loadedCells = [], choiceSet = null, presentationMode = null,
+        originalKeyboardProperties, keyboardListener = null,
+        choiceSetSelectionListener, cancelId, completionListener = null) {
+        super('_PreloadPresentChoicesOperation');
+        this._lifecycleManager = lifecycleManager;
+        this._fileManager = fileManager;
+        this._displayName = displayName;
+        this._defaultMainWindowCapability = defaultMainWindowCapability;
+        this._isVrOptional = isVrOptional;
+        this._cellsToPreload = cellsToPreload;
+        this._loadedCells = loadedCells;
+        this._completionListener = completionListener;
+        this._currentState = _PreloadPresentChoicesOperationState.NOT_STARTED;
+
+        if (choiceSet !== null) {
+            this._choiceSet = choiceSet;
+            this._choiceSet.setCanceledListener(() => {
+                this._cancelInteraction();
+            });
+            this._presentationMode = presentationMode;
+            this._originalKeyboardProperties = originalKeyboardProperties;
+            this._keyboardListener = keyboardListener;
+            this._choiceSetSelectionListener = choiceSetSelectionListener;
+            this._cancelId = cancelId;
+
+            this._sdlMsgVersion = lifecycleManager.getSdlMsgVersion();
+            this._keyboardProperties = originalKeyboardProperties;
+            this._selectedCellRow = null;
+            this._selectedCell = null;
+            this._selectedTriggerSource = null;
+            this._updatedKeyboardProperties = null;
+            this._keyboardRpcListener = null;
+        }
+        // internal usage
+        this._isRunning = false;
+    }
+
+    /**
+     * The method that causes the task to run.
+     * @param {_Task} task - The task instance
+     * @returns {Promise} - Does not resolve to any value
+     */
+    async onExecute (task) {
```

Comment:
The onExecute and _startPresent methods need to be updated for alignment. These methods should be checking the return values from methods such as preloadCells or updateKeyboardProperties and canceling the present if an error occurs along the way.